src/matrix_multiply.jl

```diff
@@ -15,11 +15,15 @@ import LinearAlgebra: BlasFloat, matprod, mul!
 @inline *(A::StaticArray{Tuple{N,1},<:Any,2}, B::Adjoint{<:Any,<:StaticVector}) where {N} = vec(A) * B
 @inline *(A::StaticArray{Tuple{N,1},<:Any,2}, B::Transpose{<:Any,<:StaticVector}) where {N} = vec(A) * B
 
+# Avoid LinearAlgebra._quad_matmul's order calculation on equal sizes
+@inline *(A::StaticMatrix{N,N}, B::StaticMatrix{N,N}, C::StaticMatrix{N,N}) where {N} = (A*B)*C
```

Comment:
You could also use `StaticMatMulLike` here to be a bit more generic, although currently `StaticMatMulLike` doesn't swap sizes for adjoints and transposes so that would have to be fixed when working on the cost model.

Reply:
I see that `StaticArrays.StaticMatMulLike{3,10}` is a union of three different sizes, is that deliberate or an oversight? Seems like it really ought to be `StaticSquareLike{3}` for the all the square ones, and `StaticMatLike{3,10}` allowing non-square but always `size(A) == (3,10)`.

Reply:
Ultimately I ended up not using the size part of `StaticMatMulLike` so the fact that `s1` and `s2` don't correspond to sizes is essentially an oversight.